code/prop/prop.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "globalincs/pstypes.h"
+
+#include "mission/mission_flags.h"
+#include "object/object.h"
+#include "object/objcollide.h"
+#include "prop/prop_flags.h"
+#include "ship/ship.h"
+
+#define MAX_PROP_DETAIL_LEVELS    MAX_SHIP_DETAIL_LEVELS
+
+typedef struct prop_info {
+	SCP_string name;                                            // Prop name
+	SCP_string category;                                        // Category name
+	SCP_string pof_file;                                        // Pof filename
+	vec3d closeup_pos;                                          // position for camera when using prop in closeup view (eg briefing and techroom)
+	float closeup_zoom;                                         // zoom when using prop in closeup view (eg briefing and techroom)
+	int model_num;                                              // The model number of the loaded POF
+	int num_detail_levels;                                      // Detail levels of the model
+	int detail_distance[MAX_PROP_DETAIL_LEVELS];                // distance to change detail levels at
+	SCP_unordered_map<int, void*> glowpoint_bank_override_map;  // Glow point bank overrides currently unused
+	flagset<Prop::Info_Flags> flags;                            // Info flags
+	SCP_map<SCP_string, SCP_string> custom_data;                // Custom data for this prop
+	SCP_vector<custom_string> custom_strings;                   // Custom strings for this prop
+} prop_info;
+
+typedef struct prop {
+	char prop_name[NAME_LENGTH];
+	int objnum;
+	int prop_info_index;
+	int model_instance_num;
+	uint create_time;                     // time prop was created, set by gettime()
+	fix time_created;
+	float alpha_mult;
+	// glow points
+	std::deque<bool> glow_point_bank_active;
```

Comment:
This should be using an SCP container.
If no SCP_deque exists, you should make one by ways of a using declaration (check how SCP_map etc. are defined).